File: android/app/src/main/java/com/motosetup/app/navigation/GlassTabBar.kt
package com.motosetup.app.navigation

import androidx.compose.animation.animateColorAsState
import androidx.compose.animation.core.Spring
import androidx.compose.animation.core.spring
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.motosetup.app.ui.theme.AppCaption
import com.motosetup.app.ui.theme.AppColor
import com.motosetup.app.ui.theme.AppSpacing
import com.motosetup.app.ui.theme.appGlass

/**
 * Tab bar custom stile "vetro": pill dietro l'icona attiva.
 * Vedi CLAUDE.md — "Navigazione — mapping presentazioni".
 */
@Composable
fun GlassTabBar(
    selected: AppTab,
    onSelect: (AppTab) -> Unit,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier
            .padding(horizontal = AppSpacing.lg)
            .appGlass(cornerRadius = 28.dp)
            .padding(6.dp),
        horizontalArrangement = Arrangement.SpaceEvenly,
    ) {
        AppTab.entries.forEach { tab ->
            TabButton(
                tab = tab,
                isSelected = tab == selected,
                onClick = { onSelect(tab) },
                modifier = Modifier.weight(1f),
            )
        }
    }
}

@Composable
private fun TabButton(
    tab: AppTab,
    isSelected: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val contentColor by animateColorAsState(
        targetValue = if (isSelected) AppColor.textPrimary else AppColor.textSecondary,
        animationSpec = spring(stiffness = Spring.StiffnessMedium),
        label = "tabContentColor",
    )
    val pillColor by animateColorAsState(
        targetValue = if (isSelected) AppColor.accentBlue.copy(alpha = 0.25f) else Color.Transparent,
        animationSpec = spring(stiffness = Spring.StiffnessMedium),
        label = "tabPillColor",
    )

    Column(
        modifier = modifier
            .clip(RoundedCornerShape(20.dp))
            .background(pillColor)
            .clickable(onClick = onClick)
            .padding(vertical = AppSpacing.sm),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Icon(
            imageVector = tab.icon,
            contentDescription = tab.title,
            tint = contentColor,
            modifier = Modifier.size(22.dp),
        )
        Text(text = tab.title, style = AppCaption, color = contentColor)
    }
}

File: MotoSetup/DesignSystem/Components/GlassTabBar.swift
import SwiftUI

enum AppTab: String, CaseIterable, Identifiable {
    case home, setup, consigliAI, profilo
    var id: String { rawValue }

    var title: String {
        switch self {
        case .home: "Home"
        case .setup: "Setup"
        case .consigliAI: "Consigli AI"
        case .profilo: "Profilo"
        }
    }

    var systemImage: String {
        switch self {
        case .home: "house.fill"
        case .setup: "magnifyingglass"
        case .consigliAI: "bubble.left.and.bubble.right.fill"
        case .profilo: "person.fill"
        }
    }
}

/// Tab bar custom stile Liquid Glass: pill animata dietro l'icona attiva.
/// Il `TabView` reale resta sotto (chrome nativo nascosto) per preservare
/// stato/scroll per tab — questa è solo la barra visuale sovrapposta.
/// Vedi CLAUDE.md — "Navigazione — mapping presentazioni".
struct GlassTabBar: View {
    @Binding var selection: AppTab
    @Namespace private var namespace

    var body: some View {
        Group {
            if #available(iOS 26, *) {
                GlassEffectContainer(spacing: 8) {
                    tabRow
                }
            } else {
                tabRow
                    .appGlass(cornerRadius: 28)
            }
        }
        .padding(.horizontal, AppSpacing.lg)
    }

    private var tabRow: some View {
        HStack(spacing: 4) {
            ForEach(AppTab.allCases) { tab in
                tabButton(tab)
            }
        }
        .padding(6)
    }

    @ViewBuilder
    private func tabButton(_ tab: AppTab) -> some View {
        let isSelected = tab == selection
        Button {
            withAnimation(.spring(response: 0.35, dampingFraction: 0.8)) {
                selection = tab
            }
        } label: {
            VStack(spacing: 2) {
                Image(systemName: tab.systemImage)
                    .font(.system(size: 20, weight: .medium))
                Text(tab.title)
                    .font(.appCaption)
            }
            .foregroundStyle(isSelected ? AppColor.textPrimary : AppColor.textSecondary)
            .padding(.vertical, 8)
            .frame(maxWidth: .infinity)
            .background(alignment: .center) {
                if isSelected {
                    selectedBackground
                        .matchedGeometryEffect(id: "tabPill", in: namespace)
                }
            }
            .contentShape(.rect)
        }
        .buttonStyle(.plain)
    }

    @ViewBuilder
    private var selectedBackground: some View {
        if #available(iOS 26, *) {
            Capsule()
                .fill(.clear)
                .glassEffect(.regular.tint(AppColor.accentBlue.opacity(0.35)), in: .capsule)
        } else {
            Capsule()
                .fill(AppColor.accentBlue.opacity(0.25))
                .background(.ultraThinMaterial, in: .capsule)
        }
    }
}

#Preview {
    ZStack(alignment: .bottom) {
        AppColor.background.ignoresSafeArea()
        GlassTabBar(selection: .constant(.home))
            .padding(.bottom, AppSpacing.lg)
    }
}
